src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -168,6 +169,9 @@ def collectAgentInfo(self):
             if float(disk['percent'].strip('%')) >= diskUseThreshold and \
                             disk['mounted'] not in self.config.AnalyticsDataCollector.ignoreDisk:
                 agentInfo['disk_warning'].append(disk)
+                agentInfo['drain_mode'] = True
+                self.config.WorkQueueManager.queueParams['DrainMode'] = True
+                os.system("$manage execute-agent wmcoreD --restart --components=WorkQueueManager,AnalyticsDataCollector,AgentStatusWatcher")
```

Comment:
So from the meeting we agree we will instead try to reread the config each polling cycle instead of actually restarting the components, a la...

https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/AgentStatusWatcher/ResourceControlUpdater.py#L89